### PR TITLE
KDEV-539: Authenticate with external identity provider

### PR DESCRIPTION
**Pull request by DennyGD**
* Add support for external identity providers:
  * modify `kinvey init` (creates a new profile) and `kinvey profile login` (gets a new token for an existing profile)

### CODE REVIEW

File: test/unit/lib/authentication.test.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        };
+
+        const sessionValue = 'testSession';
+
+        const expectedUrl = 'test.com/v4/identity-providers/1234567890/authenticate?redirect=http://localhost:5000/';
+        const openBrowserStub = sandbox.stub(Utils, 'openBrowser')
+          .withArgs(expectedUrl)
+          .callsFake(() => {
+            request({ url: `http://localhost:5000/?session=${sessionValue}` }, () => {
+              Promise.resolve();
```
Comment (georgiwe) [inline test/unit/lib/authentication.test.js:59]: What is this `Promise.resolve()` doing - it's not returned or anything?

File: lib/profile/ProfilesController.js [hunk 2]
```diff
@@ -190,6 +194,10 @@ class ProfilesController extends BaseController {
 
     async.series([
       (next) => {
+        if (profile.identityProvider.name !== 'Kinvey') {
+          return setImmediate(next);
```
[the remark this answers is not in the document]
Comment (DennyGD) [reply lib/profile/ProfilesController.js:198]: > If you get an error like RangeError: Maximum call stack size exceeded. or other stack overflow issues when using async, you are likely using a synchronous iteratee. By synchronous we mean a function that calls its callback on the same tick in the javascript event loop, without doing any I/O or using any timers. Calling many callbacks iteratively will quickly overflow the stack. If you run into this issue, just defer your callback..

[https://caolan.github.io/async/v2/](https://caolan.github.io/async/v2/)

[the remark this answers is not in the document]
Comment (georgiwe) [reply lib/profile/ProfilesController.js:198]: I know that bit but I was going off of the code I've seen when debugging. Things like [this](https://github.com/caolan/async/blob/2.x/dist/async.js#L1117). But I guess `ensureAsync()` does what I thought `wrapAsync()` does.

**Review by ivanovit — DISMISSED** [review]
LGTM. After Georgi's comments.

**Review by georgiwe — APPROVED** [review]

**Pull request merged**